crates/bevy_remote/src/lib.rs

```diff
@@ -0,0 +1,849 @@
+            let _ = message.sender.send_blocking(Err(BrpError {
+                code: error_codes::METHOD_NOT_FOUND,
+                message: format!("Method `{}` not found", message.method),
+                data: None,
+            }));
+            continue;
+        };
+
+        // Execute the handler, and send the result back to the client.
+        let result = match world.run_system_with_input(*handler, message.params) {
+            Ok(result) => result,
+            Err(error) => {
+                let _ = message.sender.send_blocking(Err(BrpError {
+                    code: error_codes::INTERNAL_ERROR,
+                    message: format!("Failed to run method handler: {error}"),
+                    data: None,
+                }));
+                continue;
+            }
+        };
+
+        let _ = message.sender.send_blocking(result);
+    }
+}
+
+/// The Bevy Remote Protocol server main loop.
+async fn server_main(
+    address: IpAddr,
+    port: u16,
+    request_sender: Sender<BrpMessage>,
+) -> AnyhowResult<()> {
+    listen(
+        Async::<TcpListener>::bind((address, port))?,
+        &request_sender,
+    )
+    .await
+}
+
+async fn listen(
+    listener: Async<TcpListener>,
+    request_sender: &Sender<BrpMessage>,
+) -> AnyhowResult<()> {
+    loop {
+        let (client, _) = listener.accept().await?;
+
+        let request_sender = request_sender.clone();
+        IoTaskPool::get()
+            .spawn(async move {
+                let _ = handle_client(client, request_sender).await;
+            })
+            .detach();
+    }
+}
+
+async fn handle_client(
+    client: Async<TcpStream>,
+    request_sender: Sender<BrpMessage>,
+) -> AnyhowResult<()> {
+    http1::Builder::new()
+        .timer(SmolTimer::new())
+        .serve_connection(
+            FuturesIo::new(client),
+            service::service_fn(|request| process_request_batch(request, &request_sender)),
+        )
+        .await?;
+
+    Ok(())
+}
+
+/// A helper function for the Bevy Remote Protocol server that handles a batch
+/// of requests coming from a client.
+async fn process_request_batch(
+    request: Request<Incoming>,
+    request_sender: &Sender<BrpMessage>,
+) -> AnyhowResult<Response<Full<Bytes>>> {
+    let batch_bytes = request.into_body().collect().await?.to_bytes();
+    let batch: Result<BrpBatch, _> = serde_json::from_slice(&batch_bytes);
+
+    let serialized = match batch {
+        Ok(batch) => match batch {
+            BrpBatch::Single(request) => {
+                serde_json::to_string(&process_single_request(request, request_sender).await?)?
+            }
+            BrpBatch::Batch(requests) => {
+                let mut responses = Vec::new();
+
+                for request in requests {
+                    responses.push(process_single_request(request, request_sender).await?);
+                }
+
+                serde_json::to_string(&responses)?
+            }
+        },
```

Comment:
Nit: You could also do
```suggestion
        Ok(BrpBatch::Single(request)) => {
            serde_json::to_string(&process_single_request(request, request_sender).await?)?
        }
        Ok(BrpBatch::Batch(requests)) => {
            let mut responses = Vec::new();

            for request in requests {
                responses.push(process_single_request(request, request_sender).await?);
            }

            serde_json::to_string(&responses)?
        }
```